### PR TITLE
feat: trino support server-cert

### PR DESCRIPTION
**Pull request by dungdm93**
Signed-off-by: Đặng Minh Dũng <dungdm93@live.com>

### SUMMARY
Add support Server certificate for Trino database

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screenshot from 2021-08-19 07-59-06

### TESTING INSTRUCTIONS
Write & pass unit-tests

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/base.py [hunk 1]
```diff
@@ -1250,6 +1249,22 @@ def get_extra_params(database: "Database") -> Dict[str, Any]:
                 raise ex
         return extra
 
+    @staticmethod
+    def get_encrypted_extra_params(database: "Database") -> Dict[str, Any]:
```
Comment (mistercrunch) [inline superset/db_engine_specs/base.py:1253]: Why moving this logic from Database model to here? At first I thought you'd be overriding it for Trino but then not, so I'm not clear why moving the logic to `BaseEngineSpec`?

File: tests/integration_tests/db_engine_specs/trino_tests.py [hunk 2]
```diff
@@ -14,11 +14,35 @@
...
+import json
+from unittest.mock import Mock
+
 from sqlalchemy.engine.url import URL
 
 from superset.db_engine_specs.trino import TrinoEngineSpec
 from tests.integration_tests.db_engine_specs.base_tests import TestDbEngineSpec
 
+TEST_CERT = """\
+-----BEGIN CERTIFICATE-----
```
Comment (mistercrunch) [inline tests/integration_tests/db_engine_specs/trino_tests.py:26]: wondering if some security automation system on our build systems at Preset is going to trigger while seeing this as it looks like stuff that shouldn't get checked into a repo

Comment (dungdm93) [reply tests/integration_tests/db_engine_specs/trino_tests.py]: Yeah. I already change from using fake `TEST_CERT` to using mock.
You should take a look

File: superset/db_engine_specs/base.py (hunk 1 above)
Comment (dungdm93) [reply superset/db_engine_specs/base.py:1253]: Actually, YES. But it's my organization specific code, so I not put it here.
But I still push the change of `get_encrypted_extra_params` in order to align with `get_extra_params`, and I think many people out there also need this.

Comment (dungdm93) [reply superset/db_engine_specs/base.py]: Hello @mistercrunch, I just move this change to another PR.
And the rest, could you please take a look.

**Comment by dungdm93** [comment]
Hello @mistercrunch, could you review this PR

**Comment by dungdm93** [comment]
Hello @mistercrunch, could you please review this one

**Review by villebro — APPROVED** [review]
LGTM, really big thanks for this added feature! Waiting for tests to pass, if all is good I think this is good to go.

**Comment by villebro** [comment]
@dungdm93 apparently there's a flaky linting rule that's failing CI - let me try to fix that

**Comment by villebro** [comment]
@dungdm93 your PR got blocked by redis typeshed stubs being updated today which was fixed here: #17555. If you rebase and remove those changes to `superset/utils/async_query_manager.py` this should be good to go.

**Comment by dungdm93** [comment]
Many thank @villebro for your support.
rebased from master and revert changes in `superset/utils/async_query_manager.py`

**Pull request merged**